### ISSUE TITLE
`docker ps --format` by Label doesn't seem to work

### ISSUE THREAD

**Issue by aaronweatherall** [docker/docker#24879]
**Output of `docker version`:**

```
Client:
 Version:      1.12.0-rc4
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   e4a0dbc
 Built:        Wed Jul 13 03:28:51 2016
 OS/Arch:      darwin/amd64
 Experimental: true

Server:
 Version:      1.12.0-rc4
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   e4a0dbc
 Built:        Wed Jul 13 03:39:43 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 20
 Running: 12
 Paused: 0
 Stopped: 8
Images: 72
Server Version: 1.12.0-rc4
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 131
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.15-boot2docker
Operating System: Boot2Docker 1.12.0-rc4 (TCL 7.2); HEAD : cbe6927 - Wed Jul 13 14:19:29 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 3.858 GiB
Name: default
ID: GJ4A:B7PK:GYMB:A7QF:BR6K:AMDV:TMDG:3I7I:FUNW:LPVG:PELX:CB6B
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 86
 Goroutines: 121
 System Time: 2016-07-21T03:06:36.198807833Z
 EventsListeners: 0
Username: aweatherall
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Docker Machine / Virtualbox.. usual setup

**Steps to reproduce the issue:**
1. `docker ps --format "{{.Label "com.docker.compose.project"}"`
2. `docker ps --format "{{.Label com.docker.compose.project}"`

Note: I've tried both based on documentation on
https://docs.docker.com/engine/reference/commandline/ps/#/formatting

**Describe the results you received:**
`template: :1: function "com" not defined`

**Describe the results you expected:**
I expect based on the documentation that I'd receive a list of containers showing only their Label for that box.

**Additional information you deem important (e.g. issue happens only occasionally):**
Basically, I'm hoping to get a list of labels for various boxes. 

Using `.Labels` returns EVERYTHING.. which is not desirable.

eg

```
com.docker.compose.container-number=1,com.docker.compose.oneoff=False,com.docker.compose.project=dockermultimachine,com.docker.compose.service=dynamodb,com.docker.compose.version=1.8.0-rc2,com.docker.compose.config-hash=92a420668ad933de6038680ba07e404fc9babf1292896343756b517422e968ea
```

**Comment by vdemeester** [comment]
@aaronweatherall `--format` is **only for formatting** the output (_it will not filter anything_). To filter the list of containers (based on labels for example), you need to use the `--filter` flag:

```
docker ps --filter label=com.docker.compose.project
```

**Comment by mikeprice99** [comment]
I've found similar results to @aaronweatherall
but it does work like this: 
```docker ps -a --format '{{.Label "com.docker.compose.project"}}'```
(btw I tested with com.docker.swarm.service.name, not com.docker.compose.project)

**Comment by thaJeztah** [comment]
@aaronweatherall @mikeprice99 I opened a pull request to explicitly add quotes to that example; https://github.com/docker/docker/pull/24892

It now shows the example like this:

<img width="862" alt="screen shot 2016-07-21 at 12 25 27" src="https://cloud.githubusercontent.com/assets/1804568/17019959/590893ee-4f3f-11e6-9419-e297d159cef3.png">

**Comment by pascalandy** [comment]
@thaJeztah 

I'm still confuse about this.

I created a bunch a monitoring services and I would like to docker ps then in one command. This is how it looks:

```
docker service create \
--name "papertrail" \
--network "$NTW_MONIT" \
--label com.about="monit" \
...
"$IMG_papertrail" 
```

How do it query services where `--label com.about="monit" `?

`docker ps -a --format XYZ ?`

Cheers!

**Comment by vdemeester** [comment]
@pascalandy services labels are not propagated to container. You would have to use `--container-label` flag instead. On the error, it will be fixed in 1.13.1 (see https://github.com/docker/docker/pull/30291).

**Comment by pascalandy** [comment]
Thank you @vdemeester ! 

```
docker service create \
--name "$CTN_papertrail" \
--network "$NTW_MONIT" \
--container-label "com.about.monit" \
...
"$IMG_papertrail" 
```

Then: 

```
docker ps --filter label=com.about.monit

CONTAINER ID        IMAGE                                                                                         COMMAND                  CREATED              STATUS              PORTS               NAMES
b453c133a3ed        gliderlabs/logspout@sha256:d608b3e72e4923a2e9e435ac4a2b79907e573b3839d7918521a15624a16b67d7   "/bin/logspout sys..."   About a minute ago   Up About a minute   80/tcp              papertrail.izjhpmacq0mafrf4l60qyfpvw.xw0s67jmo11m3b3wy5gpnep6i
```

:) So easy with real life examples :)
